app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+  validates :name, presence: true, length: {maximum: 50}
+  validates :email, presence: true,
+    length: {maximum: 255},
+    format: {with: VALID_EMAIL_REGEX},
+    uniqueness:  {case_sensitive: false}
+  validates :password, presence: true, length: {minimum: 6}
+  before_save :email_tolowercase
+  has_secure_password
+
+  private
+
+  def email_tolowercase
```

Comment:
```suggestion
  def downcase_email
```
chú ý đặt tên method bắt đầu bằng động từ